packages/components/src/Switch/Switch.tsx

```diff
@@ -72,86 +61,52 @@ export const Switch = forwardRef<HTMLInputElement, SwitchProps>(
     const { inputProps } = useSwitch(props, state, inputRef);
     const { isFocusVisible, focusProps } = useFocusRing();
     const stateProps = useStateProps({
-      checked: state.isSelected,
+      selected: state.isSelected,
       disabled: disabled,
       readOnly: readOnly,
       focus: isFocusVisible,
     });
 
-    const styles = useComponentStyles(
-      'Switch',
-      { variant, size },
-      { parts: ['container', 'label', 'track', 'thumb'] }
-    );
-
+    const classNames = useClassNames({ component: 'Switch', size, variant });
     return (
-      <Box
-        as="label"
-        __baseCSS={{
-          display: 'flex',
-          alignItems: 'center',
-          justifyContent: 'space-between',
-          gap: '1ch',
-          position: 'relative',
-          width,
-        }}
-        css={styles.container}
+      <label
+        className={cn(
+          'group/switch',
+          'relative flex items-center justify-between gap-[1ch]'
+        )}
+        style={createVar({ fieldWidth: width })}
+        {...stateProps}
       >
-        <Box
-          as="input"
+        <input
           ref={inputRef}
-          css={{
-            position: 'absolute',
-            width: '100%',
-            height: '100%',
-            top: 0,
-            left: 0,
-            zIndex: 1,
-            opacity: 0.0001,
-            cursor: inputProps.disabled ? 'not-allowed' : 'pointer',
-          }}
+          className="absolute left-0 top-0 z-[1] h-full w-full opacity-[0.0001] disabled:cursor-not-allowed"
```

Comment:
shouldn't be here for disabled the group used? @sebald @OsamaAbdellateef ? 🤔

Reply:
makes sense, yes the input should be not visible anyway.